### PR TITLE
On-preview widget fixes

### PR DESCRIPTION
**Pull request by Lawrence37**
Fixes some problems with displaying on-preview editing widgets after the enter/return key is pressed while adjusting or hovering over the widgets (#6244). Still don't now how to fix the problem with dragging a spot removal spot.

### CODE REVIEW

File: rtgui/controllines.cc [hunk 1]
```diff
@@ -260,7 +267,24 @@ bool ControlLineManager::drag1(int modifierKey)
         autoSetLineType(selected_object);
     }
 
-    return false;
+    return true;
+}
+
+void ControlLineManager::releaseEdit(void)
```
Comment (Floessie) [inline rtgui/controllines.cc:320]: @Lawrence37 If you're touching code, you're free to remove artifacts like this prehistoric `void` parameter.

File: rtgui/gradient.cc [hunk 2]
```diff
@@ -579,6 +580,19 @@ bool Gradient::drag1(int modifierKey)
     return false;
 }
 
+void Gradient::releaseEdit()
+{
+    if (lastObject >= 0) {
```
Comment (Floessie) [inline rtgui/gradient.cc:593]: It's hard to tell from the diff alone, if those `object` comparisons (`>= 0`, `> 1`, `> 0`) are correct, if "invalid" has different representations. Maybe that could be better modelled with a `std::optional<>` (which is C++17, but I have a simple C++11 implementation to share).

File: rtgui/gradient.cc [hunk 3]
```diff
@@ -579,6 +580,19 @@ bool Gradient::drag1(int modifierKey)
...
+        if (lastObject == 2 || lastObject == 3) {
```
Comment (Floessie) [inline rtgui/gradient.cc]: Magic numbers. Can't their meaning be expressed through an `enum` value?

File: rtgui/spot.cc [hunk 4]
```diff
@@ -185,6 +186,25 @@ void Spot::resetPressed()
...
+    Geometry *loGeom = getVisibleGeometryFromMO (lastObject);
+
+    if (!loGeom) {
+        EditSubscriber::action = EditSubscriber::Action::NONE;
+        return;
+    }
+
+    loGeom->state = Geometry::NORMAL;
+    sourceIcon.state = Geometry::NORMAL;
+    EditSubscriber::action = EditSubscriber::Action::NONE;
```
Comment (Floessie) [inline rtgui/spot.cc]: Can be moved up. It's the same in the `!loGeom` case.

File: rtgui/gradient.cc (hunk 2 above)
Comment (Lawrence37) [reply rtgui/gradient.cc:593]: `lastObject` gets its value from `EditDataProvider::getObject()` and I don't feel like changing that class or working around it in this PR.

File: rtgui/gradient.cc (hunk 3 above)
Comment (Lawrence37) [reply rtgui/gradient.cc]: Replaced with enum values.

File: rtgui/spot.cc (hunk 4 above)
Comment (Lawrence37) [reply rtgui/spot.cc]: Done.

**Comment by Floessie** [comment]
@Lawrence37 Much better now! Thanks! :+1:

**Pull request merged**